src/green/SphericalDiffuse.cpp

```diff
@@ -264,160 +265,140 @@ void SphericalDiffuse<ProfilePolicy>::initProfilePolicy(double e1,
-  LOG("Computing coefficient for the separation of the Coulomb singularity");
-  LOG("Computing first radial solution L = " + pcm::to_string(maxLC_));
-  TIMER_ON("computeZeta for coefficient");
-  zetaC_ = RadialFunction<StateType, LnTransformedRadial, Zeta>(
-      maxLC_, r_0_, r_infinity_, eval_, params_);
-  TIMER_OFF("computeZeta for coefficient");
-  LOG("DONE: Computing first radial solution L = " + pcm::to_string(maxLC_));
-
-  LOG("Computing second radial solution L = " + pcm::to_string(maxLC_));
-  TIMER_ON("computeOmega for coefficient");
-  omegaC_ = RadialFunction<StateType, LnTransformedRadial, Omega>(
-      maxLC_, r_0_, r_infinity_, eval_, params_);
-  TIMER_OFF("computeOmega for coefficient");
-  LOG("Computing second radial solution L = " + pcm::to_string(maxLC_));
-  LOG("DONE: Computing coefficient for the separation of the Coulomb singularity");
-
-  LOG("Computing radial solutions for Green's function");
-  TIMER_ON("SphericalDiffuse: Looping over angular momentum");
-  zeta_.reserve(maxLGreen_ + 1);
-  omega_.reserve(maxLGreen_ + 1);
-  for (int L = 0; L <= maxLGreen_; ++L) {
-    // First radial solution
-    LOG("Computing first radial solution L = " + pcm::to_string(L));
-    TIMER_ON("computeZeta L = " + pcm::to_string(L));
-    // Create an empty RadialSolution
-    RadialFunction<StateType, LnTransformedRadial, Zeta> tmp_zeta_(
-        L, r_0_, r_infinity_, eval_, params_);
-    zeta_.push_back(tmp_zeta_);
-    TIMER_OFF("computeZeta L = " + pcm::to_string(L));
-    LOG("DONE: Computing first radial solution L = " + pcm::to_string(L));
-
-    // Second radial solution
-    LOG("Computing second radial solution L = " + pcm::to_string(L));
-    TIMER_ON("computeOmega L = " + pcm::to_string(L));
-    // Create an empty RadialSolution
-    RadialFunction<StateType, LnTransformedRadial, Omega> tmp_omega_(
-        L, r_0_, r_infinity_, eval_, params_);
-    omega_.push_back(tmp_omega_);
-    TIMER_OFF("computeOmega L = " + pcm::to_string(L));
-    LOG("DONE: Computing second radial solution L = " + pcm::to_string(L));
-  }
-  TIMER_OFF("SphericalDiffuse: Looping over angular momentum");
-  LOG("DONE: Computing radial solutions for Green's function");
+    using namespace detail;
+    
+    // Parameters for the numerical solution of the radial differential equation
+    double eps_abs_ = 1.0e-10; /*! Absolute tolerance level */
+    double eps_rel_ = 1.0e-06; /*! Relative tolerance level */
+    double factor_x_ = 0.0;    /*! Weight of the state      */
+    double factor_dxdt_ = 0.0; /*! Weight of the state derivative */
+    double r_0_ = 0.1;         /*! Lower bound of the integration interval */
+    double r_infinity_ =
+        this->profile_.center() + 30.0; /*! Upper bound of the integration interval */
+    
+    double y_0_ = log(r_0_);
+    double y_infinity_ = log(r_infinity_);
+    double observer_step_ = 1.0e-2 * this->profile_.width() / this->profile_.center();     /*! Time step between observer calls */
```

Comment:
:+1: to tie the step of the integrator to the actual shape of the profile. We will have to rethink the use of `this->profile_.center()` when #107 (which makes `MembraneTanh` actually usable) is merged in.